server/core/src/rodsServer.cpp

```diff
@@ -1385,6 +1385,31 @@ int initServerMain(
 
     const auto start_delay_server = [&](std::any&)
     {
+
+
+/*  NEW 4.3.0 LEADER/SUCCESSOR ALGORITHM
+
+   - if self is leader
+     - if successor is defined
+        - if necessary, gracefully complete current jobs and exit
+     - else
+        - if necessary, start delay server
+        - grab and process delay rules, same as today
+   - else if self is successor
+     - run health check on the leader stats (utility function that calculates based on db stats)
+     - if leader is dead
+         - promote self to leader in db
+         - exit
+     - else
+         - check on the leader directly (ping?  grid-status?)
```

Comment:
ping the leader's delayserver listener port, not the main 1247?

Reply:
can get the delay server info from `irods-grid status`.   but we noticed that if the delay server was dead, the irods-grid was not informed... so might need another cron/thread to keep that information updated/fresh.